src/EditorFeatures/CSharp/BlockCommentEditing/BlockCommentEditingCommandHandler.cs

```diff
@@ -55,186 +56,199 @@ public bool ExecuteCommand(ReturnKeyCommandArgs args, CommandExecutionContext co
+                currentLine.Start,
+                firstNonWhitespacePosition));
+
+            var exteriorText = GetExteriorText();
+            if (exteriorText == null)
+                return null;
+
+            var options = document.Project.Solution.Options;
+            var newLine = options.GetOption(FormattingOptions.NewLine, LanguageNames.CSharp);
+            return newLine + exteriorText;
+
+            string GetExteriorText()
             {
-                if (BlockCommentEndsRightAfterCaret(caretPosition))
+                if (startsWithBlockCommentStartString)
                 {
-                    //      /*
-                    //      |*/
-                    return " ";
+                    if (BlockCommentEndsRightAfterCaret(caretPosition))
+                    {
+                        //      /*|*/
+                        return commentIndentation + " ";
+                    }
+                    else if (caretPosition == firstNonWhitespacePosition + 1)
+                    {
+                        //      /|*
+                        return null; // The newline inserted could break the syntax in a way that this handler cannot fix, let's leave it.
+                    }
+                    else
+                    {
+                        //      /*|
+                        // This is directly after the comment starts.  Insert ' * ' to continue the comment and to put
+                        // the user one space in.  This is the idiomatic style for C#.  Note: if the user is hitting
+                        // enter after
+                        //
+                        //  /*
+                        //   *
+                        //   *$$
+                        //
+                        // Then we don't add the space.  In this case, they are indicating they don't want this extra
+                        // space added.
+                        var padding = GetPaddingAfterCommentCharacter();
+                        return commentIndentation + " *" + (padding == "" ? " " : padding);
+                    }
                 }
-                else if (caretPosition == firstNonWhitespacePosition + 1)
+
+                if (startsWithBlockCommentEndString)
                 {
-                    //      *|/
-                    return "*";
+                    if (BlockCommentEndsRightAfterCaret(caretPosition))
+                    {
+                        //      /*
+                        //      |*/
+                        return commentIndentation + " ";
+                    }
+                    else if (caretPosition == firstNonWhitespacePosition + 1)
+                    {
+                        //      *|/
+                        return lineIndentation + "*";
+                    }
+                    else
+                    {
+                        //      /*
+                        //   |   */
+                        return commentIndentation + " ";
```

Comment:
this is much nicer than before.  this was the line that would commonly cause you to have `* */` inserted in your text.  Now we just gracefully handle this.  you'll see this in the tests.